src/main/java/uk/gov/hmcts/reform/bulkscanning/services/SasTokenGeneratorService.java

```diff
@@ -0,0 +1,95 @@
+package uk.gov.hmcts.reform.bulkscanning.services;
+
+import com.microsoft.azure.storage.StorageCredentials;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.SharedAccessBlobPermissions;
+import com.microsoft.azure.storage.blob.SharedAccessBlobPolicy;
+import org.apache.commons.lang3.EnumUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanning.config.AccessTokenConfiguration;
+import uk.gov.hmcts.reform.bulkscanning.exceptions.ServiceConfigNotFoundException;
+import uk.gov.hmcts.reform.bulkscanning.exceptions.UnableToGenerateSasTokenException;
+
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.EnumSet;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.TimeZone;
+
+@EnableConfigurationProperties(AccessTokenConfiguration.class)
+@Service
+public class SasTokenGeneratorService {
+    private final CloudBlobClient cloudBlobClient;
+    private final StorageCredentials storageCredentials;
+    private final AccessTokenConfiguration accessTokenConfiguration;
+
+    private static final Logger log = LoggerFactory.getLogger(SasTokenGeneratorService.class);
+
+    public SasTokenGeneratorService(
+        CloudBlobClient cloudBlobClient,
+        StorageCredentials storageCredentials,
+        AccessTokenConfiguration accessTokenConfiguration
+    ) {
+        this.cloudBlobClient = cloudBlobClient;
+        this.storageCredentials = storageCredentials;
+        this.accessTokenConfiguration = accessTokenConfiguration;
+    }
+
+    public String generateSasToken(String serviceName) {
+        //Based on the service name container reference would be returned for specific service
+        try {
+            log.info("SAS Token request received for service {} ", serviceName);
+
+            CloudBlobContainer cloudBlobContainer = cloudBlobClient.getContainerReference(serviceName);
+
+            return cloudBlobContainer.generateSharedAccessSignature(createSharedAccessPolicy(serviceName), null);
+        } catch (URISyntaxException | StorageException | InvalidKeyException e) {
+            throw new UnableToGenerateSasTokenException(e);
```

Comment:
`URISyntaxException` doesn't sound to me as `Internal Server Error`. Or is it something like "badly configured server with uri to container"? Then it is `Internal Server Error`